pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -582,20 +582,39 @@ private void createDerivedColumnV1Indices(String column, FunctionEvaluator funct
       int numDocs = _segmentMetadata.getTotalDocs();
       Object[] outputValues = new Object[numDocs];
       PinotDataType outputValueType = null;
+      int loggedErrorCount = 0;
+      long lastLoggedTime = System.currentTimeMillis();
+
+      FieldSpec fieldSpec = _schema.getFieldSpecFor(column);
+      Object defaultNullValue = fieldSpec.getDefaultNullValue();
       for (int i = 0; i < numDocs; i++) {
         for (int j = 0; j < numArguments; j++) {
           inputValues[j] = valueReaders.get(j).getValue(i);
         }
-        Object outputValue = functionEvaluator.evaluate(inputValues);
-        outputValues[i] = outputValue;
+        try {
+          Object outputValue = functionEvaluator.evaluate(inputValues);
+          outputValues[i] = outputValue;
+        } catch (Exception e) {
+          if (!errorOnFailure) {
+            // Try to not flush the logger, sample every 1000 errors or 10 seconds
+            if (loggedErrorCount++ % 10000 == 0 || System.currentTimeMillis() - lastLoggedTime > 10000) {
+              LOGGER.warn("Caught exception while evaluating derived column: {} with function: {} and arguments: {}",
+                  column, functionEvaluator, inputValues, e);
+              lastLoggedTime = System.currentTimeMillis();
+            }
+            outputValues[i] = defaultNullValue;
+          } else {
+            throw e;
+          }
+        }
+
         if (outputValueType == null) {
-          Class<?> outputValueClass = outputValue.getClass();
+          Class<?> outputValueClass = outputValues[i].getClass();
```

Comment:
This can cause problem when the value is default. We need to skip the type casting for default values.